draco/analysis/transform.py

```diff
@@ -889,3 +892,130 @@ def process(self, polcont):
                 outcont.datasets[name][:] *= 0.5
 
         return outcont
+
+
+class TransformJanskyToKelvin(task.SingleTask):
+    """Task to convert from Jy to Kelvin and vice-versa.
+
+    This integrates over the primary beams in the telescope class to derive the
+    brightness temperature to flux conversion.
+
+    Attributes
+    ----------
+    convert_Jy_to_K : bool
+        If True, apply a Jansky to Kelvin conversion factor. If False apply a Kelvin to
+        Jansky conversion.
+    share : {"none", "all"}
+        Which datasets should the output share with the input. Default is "all".
+    """
+
+    convert_Jy_to_K = config.Property(proptype=bool, default=True)
+    share = config.enum(["none", "all"], default="all")
+
+    def setup(self, telescope: io.TelescopeConvertible):
+        """Set the telescope object.
+
+        Parameters
+        ----------
+        telescope
+            An object we can get a telescope object from. This telescope must be able to
+            calculate the beams at all incoming frequencies.
+        """
+        self.telescope = io.get_telescope(telescope)
+        self.telescope._init_trans(256)
```

Comment:
I am not sure how important this is, but now nside is fixed. For higher observing frequencies and more cylinders you might want a higher nside parameter.

Reply:
Yeah, good point, I was modifying some of that this morning so I'll push some more changes.